build/setup.pm

```diff
@@ -419,7 +419,7 @@ our %COMPILERS = (
         ld => 'link',
         as => 'ml64',
 
-        ccmiscflags  => '/nologo /MT /std:c++latest',
+        ccmiscflags  => '/nologo /MT /std:c17 /experimental:c11atomics',
```

Comment:
It looks like `/experimental:c11atomics` causes warnings for older MSVC about not recognizing that flag. One option would be to use `/wd9002` to disable the warning, although this approach will suppress the warning for all unrecognized flags (not just `/experimental:c11atomics`)

Reply:
Hmm, looking at the CI for https://github.com/MoarVM/MoarVM/pull/1774/commits/7c2f842adbef354ca619955fc52a798c2c3853d1 I guess `/wd9002` isn't recognized either and just results in an additional warning

Reply:
I'm closing this in favor of https://github.com/MoarVM/MoarVM/pull/1777 which leaves out the `/experimental:c11atomics` flag, but makes all the other changes required to use c11 atomics on the msvc toolchain.